### PR TITLE
Implement login flow

### LINKED ISSUE #2: Set up the login screen process

### PR DESCRIPTION
**Pull request by nathan815**
I have created OauthWebViewScreen and EnterEidScreen screens.

The oauth web view currently displays EMU's login screen but it does not actually log the user in yet. Need a backend endpoint to start google oauth process which will redirect to the EMU login screen. To simulate logging in, it sets the user as logged in as soon as they click "Login" in the EMU login web page.

Closes #2.

### CODE REVIEW

File: src/screens/login/OauthWebViewScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        }
+    }, [username]);
+
+    const webViewInjectedJs = `
+        document.getElementById('username').value = '${username}';
+        document.getElementById('password').value = '';
+    `;
+
+    // TODO: use oauth url here (url from backend which will redirect to google/emich login page)
+    const oauthUrl = 'https://my.emich.edu';
```
Comment (nathan815) [inline src/screens/login/OauthWebViewScreen.tsx]: @Mazawrath  Here is where we'll put the backend url to start the oauth process once you setup an oauth endpoint (will need a python oauth library for that)

Comment (Mazawrath) [reply src/screens/login/OauthWebViewScreen.tsx]: Got it
Should I setup a compute engine server with the server running on it so you can start testing it out?

Comment (nathan815) [reply src/screens/login/OauthWebViewScreen.tsx]: Yeah that'd be good to start setting up. I prefer to test with a local server instance whenever we are testing the app locally, tho.

We should actually set up a docker image for the backend, that way its easy for anyone to run the server locally and also to deploy on compute engine.

File: src/contexts/AuthContext.tsx [hunk 2]
```diff
@@ -2,29 +2,50 @@ import React from 'react';
...
+    Logout = 'logout',
+    UpdateAuthUser = 'update auth user',
 }
 
-type Action = { type: AuthActionType, username?: string }
+export type UserInfo = {
+    username: string,
+    name?: string,
+    eid?: string,
+    imageUrl?: string,
```
[the remark this answers is not in the document]
Comment (nathan815) [reply src/contexts/AuthContext.tsx]: Yes, username is basically their email, but just the part before `@emich.edu`

> And we will ask the eid manually, right?

Correct. There is actually already a screen that asks for the EID, in this PR :)
If you checkout this branch locally (`git checkout --track origin/login-screens-2`) you can run it and see the login flow: EMU login page webview, enter EID screen, and location request screen.

**Comment by nathan815** [comment]
> What is the policy that we have for when users reject location services?
> Will users be rejected if they don't turn it on?

Originally I thought about doing it like that, but if they don't want to share their location, I think they should still be able to request SEEUS. So my thinking on this now is: If they reject location services, the map will start out in the middle of EMU's campus rather than their current location, and they will be required to select both their "Pick-Up" and Destination locations. This is similar to how Uber does it.

**Comment by Mazawrath** [comment]
I thought that the main reason why we forced them to use location services was to make sure they are actually on campus.
At this point the meeting we had with EMU police felt like years ago but I thought that they wanted us to have it for that reason, but I could be wrong.

**Comment by nathan815** [comment]
Good point. This deserves some more discussion with SEEUS supervisors and/or Sgt Torres. I don't recall exactly what they said, but what you're saying sounds familiar. I will send them an email about it.

Regardless, I think the most important verification we are doing is the login via emich email. With that, we know who is requesting it, and will be able to block them if they abuse the app. Location can be spoofed, but they can't lie about their EMU email (unless they know someone else's password of course).

**Pull request merged**